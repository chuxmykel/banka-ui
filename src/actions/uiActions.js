import {
  MODAL_OPEN,
  MODAL_CLOSE,
} from '@Actions/types';

export const openModal = modal => ({
  type: MODAL_OPEN,
  payload: modal,
});

export const closeModal = modal => ({
  type: MODAL_CLOSE,
  payload: modal,
});
