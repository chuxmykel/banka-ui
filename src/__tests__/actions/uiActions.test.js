import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { openModal, closeModal } from '@Actions/uiActions';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('UI Action tests', () => {
  const payload = 'signin';
  it('openModal should return MODAL_OPEN type with the right modal as payload', () => {
    expect(openModal(payload)).toMatchSnapshot();
  });

  it('closeModal should return MODAL_CLOSE type with the right modal as payload', () => {
    expect(closeModal(payload)).toMatchSnapshot();
  });
});

describe('Async action creators test', () => {
  const store = mockStore({});
  it('Closes a modal successfully', () => {
    return store.dispatch(closeModal('signin'))
      .then(() => {
        expect(store.getActions()).toMatchSnapshot();
      });
  });
});
