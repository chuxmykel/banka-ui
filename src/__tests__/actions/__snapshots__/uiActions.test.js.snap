// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Async action creators test Closes a modal successfully 1`] = `
Array [
  Object {
    "payload": "signin",
    "type": "MODAL_CLOSE",
  },
  Object {
    "type": "NOT_ AUTHENTICATING",
  },
  Object {
    "type": "NOT_ LOADING",
  },
]
`;

exports[`UI Action tests Should return the right types 1`] = `
Object {
  "payload": "signin",
  "type": "MODAL_OPEN",
}
`;

exports[`UI Action tests closeModal should return MODAL_CLOSE type with the right modal as payload 1`] = `[Function]`;

exports[`UI Action tests openModal should return MODAL_OPEN type with the right modal as payload 1`] = `
Object {
  "payload": "signin",
  "type": "MODAL_OPEN",
}
`;
